### PR TITLE
Implement funds type, summary section

### PR DESCRIPTION
**Pull request by mfdeveloper508**
- funds type (record 03, 16)
- summary section (record 03)
- updated transaction detail (record 16)
- fixing sample
- updated self-test

### CODE REVIEW

File: pkg/lib/account.go [hunk 1]
```diff
@@ -108,94 +100,104 @@ func (r *Account) Validate() error {
 	return nil
 }
 
-func (r *Account) Read(scan Bai2Scanner, input string, lineNum int) (int, string, error) {
+func (r *Account) Read(scan *Bai2Scanner, isRead bool) error {
```
Comment (adamdecaf) [inline pkg/lib/account.go:103]: What does `isRead` configure? The method is called `Read` already.

**Review by adamdecaf — APPROVED** [review]

File: pkg/lib/account.go (hunk 1 above)
Comment (mfdeveloper508) [reply pkg/lib/account.go]: Changed with Read(scan *Bai2Scanner, isExisted bool)

isExisted variable specify how to get line of scanner

```
for line := scan.ScanLine(isExisted); line != ""; line = scan.ScanLine(isExisted)
```
If isExisted is true, will return last line of scanner
If isExisted is false, will return new line of scanner


Read action will do on several layer like as file, group, account, detail 
isExisted is useful between read action of each layer

Comment (adamdecaf) [reply pkg/lib/account.go]: Ah okay. I think `useCurrentLine bool` is a bit clearer.

**Comment by adamdecaf** [comment]
Thanks for adding this so quickly!

**Pull request merged**